File: 2025/03/1.zig
const std = @import("std");

const fs = std.fs;
const parseInt = std.fmt.parseInt;
const ArrayList = std.ArrayList;

const print = std.debug.print;

pub fn main() !void {
    const file = try fs.cwd().openFile("./03/test.txt", .{});
    defer file.close();

    var fileBuffer: [1000]u8 = undefined;
    var reader = file.reader(&fileBuffer);

    var max: usize = 0;
    var maxI: usize = 0;
    var secMax: usize = 0;

    var sum: usize = 0;
    while (try reader.interface.takeDelimiter('\n')) |line| {
        for (0..line.len - 2) |i| {
            const array = line[i..i+1];
            print("{any}", .{array});
            const element = try std.fmt.parseInt(usize, array, 10);
            if (element > max) {
                max = element;
                maxI = i;
            }
        }
        for (maxI+1..line.len - 1) |i| {
            const array = line[i..i+1];
            const element = try std.fmt.parseInt(usize, array, 10);
            if (element > secMax) {
                secMax = element;
            }
        }
        const subSum = (10 * max) + secMax;
        print("SubSum: {d}\n", .{subSum});

        sum += subSum;
    }

    print("Sum: {d}", .{sum});
}

File: 2025/03/1.js
const fs = require('node:fs');

fs.readFile('input.txt', 'utf8', (err, data) => {
    if (err) {
        console.error(err);
        return;
    }

    let sum = 0;

    const lines = data.trim().split('\n');
    for (const line of lines) {
        let max = 0;
        let maxI = 0;
        let secMax = 0;

        for (let i = 0; i < line.length - 1; i++) {
            const num = parseInt(line[i]);
            if (num > max) {
                max = num;
                maxI = i;
            }
        }
        for (let i = maxI+1; i < line.length; i++) {
            const num = parseInt(line[i]);
            if (num > secMax) {
                secMax = num;
            }
        }

        let subSum = (max * 10) + secMax;
        sum += subSum;
    }
    console.log(sum);
});
